File: labs_26_04_18/ex1/Diamond.java
package labs_26_04_18.ex1;

// Klasa dla Rombu
public class Diamond extends Quadrangle {
    private final double side;
    private final double angle;

    public Diamond(double side, double angle) {
        this.side = side;
        this.angle = angle;
    }

    @Override
    public double calculateArea() {
        return side * side * Math.sin(Math.toRadians(angle));
    }

    @Override
    public double calculatePerimeter() {
        return 4 * side;
    }

    @Override
    public String getName() {
        return "Romb";
    }
}

File: labs_26_04_18/ex2/src/figures/diamond.cpp
#include "../../inc/diamond.hpp"
#include <cmath>

Diamond::Diamond(double s, double a) : side(s), angle(a) {}
double Diamond::calculateArea() const {
  return side * side * std::sin(angle * std::acos(-1.0) / 180.0);
}
double Diamond::calculatePerimeter() const { return 4 * side; }
std::string Diamond::getName() const { return "Romb"; }